src/scf/newton.jl

```diff
@@ -0,0 +1,162 @@
+# Newton's algorithm for the direct minimization of the energy
+
+"""
+    newton_step(basis::PlaneWaveBasis, φ, φproj, res, occ;
+                tol_krylov=1e-12, krylov_verb=1)
+
+Perform a newton step : we take as given a planewave set φ and we return the
+newton step δφ and the updated state φ - δφ
+(after proper orthonormalization) where δφ solves Jac * δφ = res
+and Jac is the Jacobian of the projected gradient descent : Jac = Ω+K.
+(cf. Eric Cancès, Gaspard Kemlin, Antoine Levitt. Convergence analysis of
+direct minimization and self-consistent iterations. SIAM Journal of Matrix
+Analysis and Applications, 42(1):243–274 (2021))
+"""
+function newton_step(basis::PlaneWaveBasis, φ, φproj, res, occ;
+                     tol_krylov=1e-12, krylov_verb=1)
+
+    # necessary quantities
+    N = size(φ[1],2)
+    Nk = length(basis.kpoints)
+    ortho(ψk) = Matrix(qr(ψk).Q)
```

Comment:
I would have these functions outside the function. Also try to merge with direct_minimization

Reply:
Same remark than https://github.com/JuliaMolSim/DFTK.jl/pull/433#discussion_r642997828

Reply:
Okay, added in src/common to be used in both newton.jl and direct_minimization.jl, but with name ortho_qr, otherwise there are conflicts with other packages.